File: 999-available-captures-for-rook.java
class Solution {
    public int numRookCaptures(char[][] board) {
        int rookR = -1, rookC = -1;
        for (int r = 0; r < 8; r++)
            for (int c = 0; c < 8; c++)
                if (board[r][c] == 'R') {
                    rookR = r;
                    rookC = c;
                }

        int captures = 0;
        int[][] directions = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};

        for (int[] d : directions) {
            int nr = rookR + d[0], nc = rookC + d[1];
            while (nr >= 0 && nr < 8 && nc >= 0 && nc < 8) {
                if (board[nr][nc] == 'B') break;
                if (board[nr][nc] == 'p') {
                    captures++;
                    break;
                }
                nr += d[0];
                nc += d[1];
            }
        }
        return captures;
    }
}   

File: 999-available-captures-for-rook.py
class Solution:
    def numRookCaptures(self, board):
        # Find the rook
        for r in range(8):
            for c in range(8):
                if board[r][c] == 'R':
                    rook_r, rook_c = r, c

        captures = 0

        # Four directions: up, down, left, right
        directions = [
            (-1, 0),
            (1, 0),
            (0, -1),
            (0, 1)
        ]

        for dr, dc in directions:
            nr, nc = rook_r + dr, rook_c + dc

            while 0 <= nr < 8 and 0 <= nc < 8:
                if board[nr][nc] == 'B':
                    break

                if board[nr][nc] == 'p':
                    captures += 1
                    break

                nr += dr
                nc += dc

        return captures
